lib/cli/ui/prompt.rb

```diff
@@ -88,6 +88,30 @@ def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true, m
           end
         end
 
+        # Asks the user for a single-line answer, without displaying the characters while typing.
+        # Typically used for password prompts
+        #
+        # ==== Return Value
+        #
+        # The password, without a trailing newline.
+        # If the user simply presses "Enter" without typing any password, this will return an empty string.
+        def ask_password(question)
+          require 'io/console'
+
+          CLI::UI.with_frame_color(:blue) do
+            STDOUT.print(CLI::UI.fmt('{{?}} ' + question))
+
+            # noecho interacts poorly with Readline under system Ruby, so do a manual `gets` here.
+            # No fancy Readline integration (like echoing back) is required for a password prompt anyway.
+            password = STDIN.noecho do
+              STDIN.gets.chomp
```

Comment:
May be pedantic, but what about people with spaces at the beginning or the end of their password?

Reply:
That is not what I thought chomp did, my bad.

Reply:
Chomp will remove the one new line character added by `gets`, without touching potential extra spaces: `" 123 \n".chomp => " 123 "`
Added a comment

Reply:
irb is great for checking these things:
```ruby
$ irb
irb(main):001:0> puts "  d  \n".chomp.inspect
"  d  "
=> nil
```

Reply:
Gah.  Stupid github updating with one comment but not the others.